wallet/src/main/scala/org/bitcoins/wallet/models/SpendingInfoDAO.scala

```diff
@@ -99,9 +101,13 @@ case class SpendingInfoDAO()(
     database.runVec(filtered.result)
   }
 
-  /** Enumerates all unspent TX outputs in the wallet */
+  private val receivedStates: Vector[TxoState] =
+    Vector(TxoState.UnconfirmedReceived, TxoState.ConfirmedReceived)
+
+  /** Enumerates all unspent TX outputs in the wallet with the state
+    * [[TxoState.UnconfirmedReceived]] or [[TxoState.ConfirmedReceived]] */
   def findAllUnspent(): Future[Vector[SpendingInfoDb]] = {
-    val query = table.filter(!_.spent)
+    val query = table.filter(_.state.inSet(receivedStates))
```

Comment:
It might be useful to have one that gets on the confirmed if someone doesn't want to try and spend unconfirmed txs

Reply:
I think this is useful, but I'll let this go into a separate PR. I want to try and keep this one as small as possible.